### PR TITLE
Backend JAX supports model.predict

### PR DESCRIPTION
**Pull request by ZongrenZou**

### CODE REVIEW

File: deepxde/model.py [hunk 1]
```diff
@@ -935,6 +935,24 @@ def op(inputs):
             # Clear cached Jacobians and Hessians.
             grad.clear()
             y = utils.to_numpy(y)
+        elif backend_name == "jax":
+            if utils.get_num_args(operator) == 2:
+
+                @jax.jit
+                def op(inputs):
+                    y_fn = lambda _x: self.net.apply(self.params[0], _x)
```
Comment (lululxvi) [inline deepxde/model.py:943]: `self.params[0]` --> `self.net.params`

File: deepxde/model.py [hunk 2]
```diff
@@ -935,6 +935,24 @@ def op(inputs):
...
+                    return operator(inputs, (y_fn(inputs), y_fn))
```
Comment (lululxvi) [inline deepxde/model.py:944]: Do you need vmap for y_fn(inputs)?

File: deepxde/model.py [hunk 3]
```diff
@@ -935,6 +935,24 @@ def op(inputs):
...
+
+            elif utils.get_num_args(operator) == 3:
+                # TODO: JAX backend Implementation of Auxiliary variables.
+                raise NotImplementedError(
+                    "Model.predict() with auxiliary variable hasn't been implemented "
+                    "for backend jax."
+                )
+            y = op(x)
+            # Clear cached Jacobians and Hessians.
+            grad.clear()
```
Comment (lululxvi) [inline deepxde/model.py:954]: JAX doesn't need grad.clear

Comment (ZongrenZou) [reply deepxde/model.py]: Done.

File: deepxde/model.py (hunk 2 above)
Comment (ZongrenZou) [reply deepxde/model.py:944]: No. flax supports vector-valued input. As far as I know, current deepxde only has vmap in "gradients".

File: deepxde/model.py (hunk 1 above)
Comment (ZongrenZou) [reply deepxde/model.py]: Yes this is better. Done.
